### PR TITLE
Radiation: decorrelation length overlap option

### PR DESCRIPTION
**Pull request by brianhenn**
Adds the decorrelation-length cloud overlap option to the `radiation` package's port of the GFS radiation scheme. Also added [basic sphinx docs](http://vulcanclimatemodeling.com/docs/fv3net/readme_links/radiation_readme.html) for the radiation package to our site.

Significant internal changes:
- `iovrsw`/`iovrlw` can now be `3` in the `radiation.RadiationConfig` class, causing the scheme to run with decorrelation length cloud overlap

- [X] Tests added

Coverage reports (updated automatically):
- test_unit: [60%](https://output.circle-artifacts.com/output/job/6f1dfb7d-4ebe-42f2-a073-4629777008ca/artifacts/0/tmp/coverage/htmlcov-test_unit/index.html)

### CODE REVIEW

File: external/radiation/tests/test_wrapper_api.py [hunk 1]
```diff
@@ -115,25 +115,27 @@ def __init__(self):
 }
 
 
-def test_config_validation(regtest):
```
Comment (brianhenn) [inline external/radiation/tests/test_wrapper_api.py:118]: This is mostly just better organization of tests that accomplish the same things

File: external/radiation/radiation/radlw/radlw_main.py [hunk 2]
```diff
@@ -85,14 +88,29 @@ def mcica_subcol(iovrlw, cldf, nlay, dz, de_lgth, iplon, rand2d):
                 else:
                     cdfunc[n, k] = cdfunc[n, k] * tem1
 
+    if iovrlw == 3:  # decorrelation length overlap
+
+        fac_lcf = np.exp(-0.5 * (dz[1:] + dz[:-1]) / de_lgth)
```
Comment (brianhenn) [inline external/radiation/radiation/radlw/radlw_main.py:93]: Could add a comment that this follows directly from [here](https://github.com/ai2cm/fv3gfs-fortran/blob/2a3ea739723d6152fd03d419c76d25741f7fa9da/FV3/gfsphysics/physics/radlw_main.f#L1967)

**Review by oliverwm1 — APPROVED** [review]
Approving, although there were a couple things I was unclear on.

File: external/radiation/radiation/radlw/radlw_main.py [hunk 3]
```diff
@@ -85,14 +88,29 @@ def mcica_subcol(iovrlw, cldf, nlay, dz, de_lgth, iplon, rand2d):
...
+        tem1 = 1.0 - cldf[k]
+
+        for n in range(ngptlw):
+            lcloudy[n, k] = cdfunc[n, k] >= tem1
 
     return lcloudy
 
 
-@jit(nopython=True)
+# @jit(nopython=True)
```
[the remark this answers is not in the document]
Comment (brianhenn) [reply external/radiation/radiation/radlw/radlw_main.py:113]: Nope, thanks good catch.

File: external/radiation/radiation/radlw/radlw_main.py [hunk 4]
```diff
@@ -58,16 +54,23 @@ def mcica_subcol(iovrlw, cldf, nlay, dz, de_lgth, iplon, rand2d):
...
 
     #  --- ...  sub-column set up according to overlapping assumption
 
+    if iovrlw in (0, 1, 3):  # random, max-random or decorr overlap
+
+        k1 = 0
+        for n in range(ngptlw):
+            for k in range(nlay):
+                cdfunc[n, k] = rand2d[k1]
+                k1 = k1 + 1
```
[the remark this answers is not in the document]
Comment (brianhenn) [reply external/radiation/radiation/radlw/radlw_main.py:72]: I think in the optimizing this summer the code for options 0 and 1 ended up doing the same thing different ways (`np.reshape` vs looping) in LW vs SW. So this just restored that both use the same way, but shouldn't change the result.

**Pull request merged**